Updating freshness ref page (#4770)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-properties/freshness.md

In the `freshness` block, one or both of `warn_after` and `error_after` can be provided. If neither is provided, then dbt will not calculate freshness snapshots for the tables in this source.

<VersionBlock firstVersion="1.7">

If a source has a `freshness:` block, dbt will attempt to calculate freshness for that source:
- If a `loaded_at_field` is provided, dbt will calculate freshness via a select query (behavior prior to v1.7).
- If a `loaded_at_field` is _not_ provided, dbt will calculate freshness via warehouse metadata tables when possible (new in v1.7).

To exclude a source from freshness calculations, you have two options:
- Don't add a `freshness:` block.
- Explicitly set `freshness: null`.

Freshness blocks are applied hierarchically:
- A `freshness` property added to a source will be applied to all tables defined in that source.
- A `freshness` property added to a source _table_ will override any properties applied to the source.

> This is useful when all of the tables in a source have the same `loaded_at_field`.

Author: ```suggestion
```

In the `freshness` block, one or both of `warn_after` and `error_after` can be provided. If neither is provided, then dbt will not calculate freshness snapshots for the tables in this source.

<VersionBlock firstVersion="1.7">

> Source freshness is calculated for all tables by default.

Author: ```suggestion
```

If a source has a `freshness:` block, dbt will attempt to calculate freshness for that source:
- If a `loaded_at_field` is provided, dbt will calculate freshness via a select query (behavior prior to v1.7).
- If a `loaded_at_field` is _not_ provided, dbt will calculate freshness via warehouse metadata tables when possible (new in v1.7).

To exclude a source from freshness calculations, you have two options:
- Don't add a `freshness:` block.
- Explicitly set `freshness: null`.

Freshness blocks are applied hierarchically:

> - A `freshness` property added to a source will be applied to all all tables defined in that source

Author: ```suggestion
- A `freshness` property added to a source will be applied to all tables defined in that source.
```

- If a `loaded_at_field` is provided, dbt will calculate freshness via a select query (behavior prior to v1.7).
- If a `loaded_at_field` is _not_ provided, dbt will calculate freshness via warehouse metadata tables when possible (new in v1.7 on supported adapters).

Currently, calculating freshness from warehouse metadata tables is supported on:
- [Snowflake](/reference/resource-configs/snowflake-configs)

Support is coming soon to the following adapters:
- [Redshift](/reference/resource-configs/redshift-configs)
- [BigQuery](/reference/resource-configs/bigquery-configs)
- [Spark](/reference/resource-configs/spark-configs)

Freshness blocks are applied hierarchically:
- a `freshness` and `loaded_at_field` property added to a source will be applied to all all tables defined in that source
- a `freshness` and `loaded_at_field` property added to a source _table_ will override any properties applied to the source.

This is useful when all of the tables in a source have the same `loaded_at_field`, as is often the case.

To exclude a source from freshness calculations, you have two options:
- Don't add a `freshness:` block.
- Explicitly set `freshness: null`.

## loaded_at_field

> (Required. Optional on adaters that support pulling freshness from warehouse metadata tables.)

Reviewer: ```suggestion
(Optional on adapters that support pulling freshness from warehouse metadata tables, required otherwise.)
```
